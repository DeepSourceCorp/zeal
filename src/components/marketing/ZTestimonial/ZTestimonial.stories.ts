import ZTestimonial from './ZTestimonial.vue'

export default {
  title: 'Zeal Marketing/Testimonial',
  component: ZTestimonial,
  excludeStories: /.*Data$/
}

export const Basic = () => ({
  components: { ZTestimonial },
  template: `<div class='padded-container w-1/2'>
      <z-testimonial class="bg-ink-300">
        <template slot="image">
          <img src="https://i.imgur.com/ISsHpNY.png" class="object-contain"/>
        </template>
        <template slot="body">
          DeepSource actually helped us resolve more than 50 bugs, which has made the code base much more stable and dependable. It allowed us to identify many more areas for improvement.
        </template>
        <template slot="footer">
          Piero Molino, Senior Technology Officer
        </template>
      </z-testimonial>
    </div>`
})
