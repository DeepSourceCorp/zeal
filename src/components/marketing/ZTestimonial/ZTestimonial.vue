<template>
  <div
    class="
      rounded-sm
      md:rounded-md
      overflow-hidden
      transition-all
      shadow
      flex flex-col
      justify-around
      h-full
      px-5
      py-4
      md:px-10 md:py-8
    "
  >
    <div class="flex-1" v-if="$slots.image">
      <slot name="image"> </slot>
    </div>
    <span class="flex-1">
      <div class="text-vanilla-100 text-md md:text-xl mb-6 pr-2" v-if="$slots.body">
        <slot name="body"> </slot>
      </div>
      <div class="text-vanilla-400 text-md md:text-lg" v-if="$slots.footer">
        <slot name="footer"> </slot>
      </div>
    </span>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'

export default Vue.extend({
  name: 'ZTestimonial'
})
</script>
