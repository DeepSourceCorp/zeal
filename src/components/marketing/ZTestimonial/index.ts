import ZTestimonial from './ZTestimonial.vue'

export default ZTestimonial
