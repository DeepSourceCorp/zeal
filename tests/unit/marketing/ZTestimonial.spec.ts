import Vue from 'vue'
import { createLocalVue, mount, Wrapper } from '@vue/test-utils'

import ZTestimonial from '../../../src/components/marketing/ZTestimonial'

describe('Testimonial component', () => {
  let mountFunction: (options?: Record<string, unknown>) => Wrapper<Vue>
  let localVue: typeof Vue
  beforeEach(() => {
    localVue = createLocalVue()
    mountFunction = (options = {}) => {
      return mount(ZTestimonial, {
        localVue,
        ...options
      })
    }
  })
  it('renders a testimonial card when the component is mounted', () => {
    expect(mountFunction().html()).toMatchSnapshot()
  })

  it('renders slots only when they are available', () => {
    let mountedInstance = mountFunction({
      slots: {
        image: ['<img id="image">']
      }
    })
    expect(mountFunction().html()).toMatchSnapshot()
    expect(mountedInstance.findAll('#image').length).toEqual(1)
    expect(mountedInstance.findAll('#body').length).toEqual(0)
    expect(mountedInstance.findAll('#footer').length).toEqual(0)

    mountedInstance = mountFunction({
      slots: {
        image: ['<img id="image">'],
        body: ['<img id="body">'],
        footer: ['<img id="footer">']
      }
    })
    expect(mountedInstance.findAll('#image').length).toEqual(1)
    expect(mountedInstance.findAll('#body').length).toEqual(1)
    expect(mountedInstance.findAll('#footer').length).toEqual(1)
  })
})
