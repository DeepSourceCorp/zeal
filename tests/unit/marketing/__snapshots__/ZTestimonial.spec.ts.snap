// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testimonial component renders a testimonial card when the component is mounted 1`] = `
<div class="rounded-xl overflow-hidden transition-all shadow flex flex-col justify-around h-full px-5 py-4 md:px-10 md:py-8">
  <!----> <span class="flex-1"><!----> <!----></span>
</div>
`;

exports[`Testimonial component renders a testimonial card when the component is mounted 2`] = `
<div class="rounded-xl overflow-hidden transition-all shadow flex flex-col justify-around h-full px-5 py-4 md:px-10 md:py-8">
  <!----> <span class="flex-1"><!----> <!----></span>
</div>
`;

exports[`Testimonial component renders slots only when they are available 1`] = `
<div class="rounded-xl overflow-hidden transition-all shadow flex flex-col justify-around h-full px-5 py-4 md:px-10 md:py-8">
  <!----> <span class="flex-1"><!----> <!----></span>
</div>
`;

exports[`Testimonial component renders slots only when they are available 2`] = `
<div class="rounded-xl overflow-hidden transition-all shadow flex flex-col justify-around h-full px-5 py-4 md:px-10 md:py-8">
  <!----> <span class="flex-1"><!----> <!----></span>
</div>
`;
